packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -103,7 +103,12 @@ abstract class ValueListenable<T> extends Listenable {
 ///  * [ValueNotifier], which is a [ChangeNotifier] that wraps a single value.
 class ChangeNotifier implements Listenable {
   int _count = 0;
-  List<VoidCallback?> _listeners = List<VoidCallback?>.filled(0, null);
+  // The _listeners is intentionally set to a fixed-length _GrowableList instead
+  // of const [] for performance reasons.
+  // See https://github.com/flutter/flutter/pull/71947/files#r545722476 for
```

Comment:
can we inline the explanation here rather than linking to github? the code will probably outlive that link.